### DISCUSSION TITLE
JSON Rendering

### DISCUSSION THREAD

**Discussion by lpe** [avo-hq/avo#1405] [Feature requests]
It would be very usefull to be able to use / render JSON in controllers (CRUD methods).

**Comment by adrianthedev** [comment]
I'm not sure that this is so trivial. 
I mean, what would you get? All records? 
Maybe you need them filtered, apply a scope, set them in an order. Right?

**Comment by lpe** [comment]
Just the ability to have the actual methods responding to json (essentially GET). So if we need to fetch records in JS we don't have to build our own controllers / methods ? Ex : /avo/resources/contacts/6 => /avo/resources/contacts/6.json

**Reply by adrianthedev** [reply]
That's "easy" to do for single records, but for lists, that's a different story. You have to tackle pagination, filtering, ordering, normalization, and lots more.

I think the easiest way right now is to have your own controller where you use Rails' native REST & JSON capabilities.

We're thinking about adding something like that in v3.0. Maybe we can collaborate and come up with solutions together [here](https://github.com/avo-hq/avo/issues/1146).
